src/formatDate.js

```diff
@@ -50,7 +50,32 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArray = date.split(fromFormat[3]);
+  const currentDate = {};
+
+  for (let i = 0; i < 3; i++) {
+    currentDate[fromFormat[i]] = dateArray[i];
+  }
+
+  if (currentDate.hasOwnProperty('YY')) {
+    currentDate.YYYY = currentDate.YY;
+
+    if (currentDate.YYYY < 30) {
+      currentDate.YYYY = '20' + currentDate.YY;
+    } else {
+      currentDate.YYYY = '19' + currentDate.YY;
+    }
+  } else {
+    currentDate.YY = currentDate.YYYY.slice(-2);
+  }
+
+  const newFormatDate = toFormat.slice(0, 3);
```

Comment:
Why don’t you want to create here empty array and use toFormat to set it with elements on 75 line?